### ISSUE TITLE
position not retained when coming back from another fragment

### ISSUE THREAD

**Issue by palicka** [tommybuonomo/dotsindicator#68]
Hi, I use navigation graph in my app and when I leave fragment which contains viewpager and come back (click on "skip" button in attached gif), viewpager's position is retained, but dots indicator is reset. I know that view is recreated when leaving and coming back, but I noticed ViewPager2.OnPageChangeCallback's onPageSelected is called  with last position, but dots indicator somehow doesn't react. But when I scroll , indicator jumps to correct position. Please have a look at attached video.  Thanks 
![ezgif-1-d6bbe5d48bf5](https://user-images.githubusercontent.com/6105846/69889141-b0becf00-12ef-11ea-9535-5208bf09c635.gif)

this is how it's initialised in onActivityCreated() method:

```
        val pagerAdapter = MyViewPagerAdapter(activity!!)
        viewPager.adapter = pagerAdapter
        pagerIndicator.setViewPager2(viewPager)
```

**Comment by VladPalamarchuk** [comment]
@palicka I had the same issue, look my solution, hope it will helps 
`
indicator.setViewPager(pager)
pager.post { adapter.notifyDataSetChanged() }
`
Where indicator - is indicator view, pager - is ViewPager, adapter - is PagerAdapter. Indicator internally react on adapter changes, so it will redraw indicators correctly. Hope it helps :)

**Comment by palicka** [comment]
@VladPalamarchuk thank you :) it works, but it would great if it was on that position right away :) but your solution is better than nothing so thank you again 👍

**Comment by ryandt** [comment]
This problem also exists if you rotate the device. `ViewPager2` will continue to show the current `Fragment` but the indicator will reset to the first dot even if the first `Fragment` isn't showing.

**Comment by Natkarock** [comment]
This bug is still existing in version 4.1.2

**Comment by tommybuonomo** [comment]
Fixed in `4.3`

**Comment by tommybuonomo** [comment]
Hello, thanks for your contribution !
This issue is now fixed in the new version `4.3`
Thanks 🔥
